src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -121,6 +135,21 @@ private void internalUsersApiRequestHandlers(RequestHandler.RequestHandlersBuild
                     .map(this::validateAndUpdatePassword)
                     .map(this::addEntityToConfig)
             );
+
+    }
+
+    protected final ValidationResult<SecurityConfiguration> filterUsers(SecurityDynamicConfiguration users, String userType) {
+        userService.includeAccountsIfType(users, AccountType.fromString(userType));
+        return ValidationResult.success(SecurityConfiguration.of(users.getCType().toString(), users));
+
+    }
+
+    protected final String filterParam(final RestRequest request) {
+        final String name = request.param("filterBy");
+        if (Strings.isNullOrEmpty(name)) {
+            return "any";
+        }
```

Comment:
Why not using the enum?

Reply:
Yep, it goes with your other comment, the enum represents the account type. I will change the enum to be  "UserFilterType"